Assets/MixedRealityToolkit.SDK/Experimental/HandCoach/Scripts/MoveToTarget.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.Collections;
+using System.Net.NetworkInformation;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    public class MoveToTarget : MonoBehaviour
+    {
+        [Tooltip("Object to track")]
+        [SerializeField]
+        private GameObject m_trackingObject = null;
+
+        [Tooltip("Target to move to")]
+        [SerializeField]
+        private GameObject m_targetObject = null;
+
+        [Tooltip("Shared parent between tracking and target objects used for relative local positions")]
+        [SerializeField]
+        private GameObject m_rootObject = null;
+
+        [Tooltip("Duration of move from tracking object to target object")]
+        [SerializeField]
+        private float m_duration = 1.38f;
+
+        [Tooltip("Tunable offset to get the GameObject to arrive at the right target position")]
+        [SerializeField]
+        private Vector3 m_targetOffset = new Vector3(0.05f, -0.1f, -0.2f);
+
+        [Tooltip("Lerp curve")]
+        [SerializeField]
+        private AnimationCurve m_animationCurve = AnimationCurve.Linear(0f, 0f, 1f, 1f);
+
+        // The local position of this gameObject
+        private Vector3 m_position;
+
+        // The local position of m_targetObject
+        private Vector3 m_targetPosition;
+
+        // The offset from this gameObject's position to the m_trackingObject's position
+        private Vector3 m_trackingOffset;
+
+        // bool to determine when to stop the follow sequence
+        private bool m_followingTargetObject;
+
+        private void Awake()
+        {
+            if (m_trackingObject != null)
+            {
+                m_trackingOffset = new Vector3(0,0,0); // GetRelativeLocalPosition(m_trackingObject, m_rootObject) - gameObject.transform.parent.localPosition;
+            }
+        }
+
+        // Since this script can attach to an object with an animator, we need to update position in LateUpdate
+        private void LateUpdate()
+        {
+            if (m_targetObject != null && m_rootObject != null)
+            {
+                m_targetPosition = GetRelativeLocalPosition(m_targetObject, m_rootObject) + m_targetOffset;
+                transform.parent.localPosition = m_position;
```

Comment:
this script needs to be on the same gameObject as the animator. We're changing the parent of this GameObject (where InteractionHint lives) so that the animations can still do their thing and change position of the rig and its components.

Reply:
would you mind adding a comment to this just in case someone else gets confused by this